### PR TITLE
Peer Reviewed Schema Changes

### PR DESCRIPTION
**Pull request by joedevivo**

### CODE REVIEW

File: priv/riak_core.schema [hunk 1]
```diff
@@ -1,7 +1,8 @@
 %%-*- mode: erlang -*-
 %% Default Bucket Properties
 
-%% @doc the number of replicas stored. Note: See CAP Controls for further discussion.
+%% @doc the number of replicas stored. Note: See Replication
```
Comment (seancribbs) [inline priv/riak_core.schema]: Can we capitalize the initial word?

File: priv/riak_core.schema [hunk 2]
```diff
@@ -205,8 +54,8 @@
   {commented, "{{platform_etc_dir}}/cert.pem"}
 ]}.
 
-%% @doc Default key location for https can be overridden
-%% with the ssl config variable, for example:
+%% @doc Default key location for https can be overridden with the ssl
+%% config variable, for example:
 {mapping, "ssl.keyfile", "riak_core.ssl.keyfile", [
   {commented, "{{platform_etc_dir}}/key.pem"}
```
Comment (seancribbs) [inline priv/riak_core.schema:63]: This should have a datatype, probably 'file'.

File: priv/riak_core.schema [hunk 3]
```diff
@@ -205,8 +54,8 @@
   {commented, "{{platform_etc_dir}}/cert.pem"}
```
Comment (seancribbs) [inline priv/riak_core.schema:56]: Datatype = file

Comment (seancribbs) [inline priv/riak_core.schema:56]: Also for the ring.state_dir, datatype = directory

File: priv/riak_core.schema [hunk 4]
```diff
@@ -221,70 +70,62 @@
...
 %% @doc To encrypt riak_core intra-cluster data handoff traffic,
-%% uncomment the following line and edit its path to an
-%% appropriate certfile and keyfile.  (This example uses a
-%% single file with both items concatenated together.)
+%% uncomment the following line and edit its path to an appropriate
+%% certfile and keyfile.  (This example uses a single file with both
+%% items concatenated together.)
 {mapping, "handoff.ssl.certfile", "riak_core.handoff_ssl_options.certfile", [
-  {commented, "/tmp/erlserver.pem"}
+%%  {commented, "/tmp/erlserver.pem"},
```
Comment (seancribbs) [inline priv/riak_core.schema:86]: datatype = file

File: priv/riak_core.schema [hunk 5]
```diff
@@ -221,70 +70,62 @@
...
+  {level, advanced}
 ]}.
 
 %% @doc if you need a seperate keyfile for handoff
-{mapping, "handoff.ssl.keyfile", "riak_core.handoff_ssl_options.keyfile", []}.
+{mapping, "handoff.ssl.keyfile", "riak_core.handoff_ssl_options.keyfile", [
+  {level, advanced}
```
Comment (seancribbs) [inline priv/riak_core.schema:94]: datatype = file

File: priv/riak_core.schema [hunk 6]
```diff
@@ -221,70 +70,62 @@
...
-}.
-
 %% Platform-specific installation paths (substituted by rebar)
 {mapping, "platform_bin_dir", "riak_core.platform_bin_dir", [
   {default, "{{platform_bin_dir}}"}
 ]}.
 
+%% @see platform_bin_dir
 {mapping, "platform_data_dir", "riak_core.platform_data_dir", [
   {default, "{{platform_data_dir}}"}
```
Comment (seancribbs) [inline priv/riak_core.schema:115]: platform_*_dir should all have datatype directory

File: priv/riak_core.schema (hunk 1 above)
Comment (joedevivo) [inline priv/riak_core.schema]: You mean "The"?

Comment (seancribbs) [inline priv/riak_core.schema]: Indeed.

File: priv/riak_core.schema [hunk 7]
```diff
@@ -196,24 +45,28 @@
...
 %% @doc Default cert location for https can be overridden
 %% with the ssl config variable, for example:
 {mapping, "ssl.certfile", "riak_core.ssl.certfile", [
+  {datatype, file},
   {commented, "{{platform_etc_dir}}/cert.pem"}
 ]}.
 
-%% @doc Default key location for https can be overridden
-%% with the ssl config variable, for example:
+%% @doc Default key location for https can be overridden with the ssl
```
Comment (seancribbs) [inline priv/riak_core.schema:59]: This docstring sucks. Idea:

```
The file containing the key used to encrypt SSL/TLS client connections.
```

On the other hand, knowing what this does tells me it's in the wrong application.

File: priv/riak_core.schema [hunk 8]
```diff
@@ -196,24 +45,28 @@
...
+%% config variable, for example:
 {mapping, "ssl.keyfile", "riak_core.ssl.keyfile", [
+  {datatype, file},
   {commented, "{{platform_etc_dir}}/key.pem"}
 ]}.
 
 %% @doc Default signing authority location for https can be overridden
```
Comment (seancribbs) [inline priv/riak_core.schema:66]: Again, maybe this belongs in `riak_api`, even if it generates a `riak_core` setting.

File: priv/riak_core.schema (hunk 4 above)
Comment (seancribbs) [inline priv/riak_core.schema:86]: . @lucperkins also mentioned this setting, perhaps we could make it point to something more sensible even though it is commented? e.g.

``` erlang
{commented, "{{platform_etc_dir}}/handoff.pem"}
```

Comment (seancribbs) [inline priv/riak_core.schema:86]: Meh, the keyfile in the next setting omits a default. Let's just remove the line.

**Comment by seancribbs** [comment]
I retract my previous comments, `riak_core.ssl` has to stay because it is used by `riak_repl` and potentially others.

**Comment by seancribbs** [comment]
`one:plus().`

**Pull request merged**